### DISCUSSION TITLE
iOS Flet App: allow running local Python files (not loading from network)

### DISCUSSION THREAD

**Discussion by yhcharles** [flet-dev/flet#1840] [General]
**Please Describe The Problem To Be Solved**

Currently, in order to run a Python file in the iOS Flet App, we need to get a host to start a service, and connect to this service in the iOS Flet App.
Can we allow users to run from a local Python file stored on device, similar to the [Pythonista iOS App](https://apps.apple.com/us/app/pythonista-3/id1085978097)?

**(Optional): Suggest A Solution**

It will be great to have this feature. One concern might be the iOS policy, but given there are other Apps (like Pythonista) that already have this capability, I think it is possible.

**Comment by FeodorFitsner** [comment]
Check these examples: https://github.com/flet-dev/serious-python/tree/main/example

**Reply by yhcharles** [reply]
Thanks for the response. Let me try to make my question more clear:
My suggestion is to update the current iOS Flet app, so that it also allows us to run a Python file stored on device. For example, users can write their Flet app on laptop, then transfer it to mobile device. And they can start it from within the iOS Flet app, similar to what Pythonista can do: https://apps.apple.com/us/app/pythonista-3/id1085978097

The benefit is users can run their program without network, and no need to maintain a server to host the app.
